module/zcommon/zfs_fletcher.c

```diff
@@ -380,18 +380,52 @@ void
 fletcher_4_incremental_native(const void *buf, uint64_t size,
     zio_cksum_t *zcp)
 {
+	const fletcher_4_ops_t *ops;
+	uint64_t p2size = P2ALIGN(size, 64);
+
 	ASSERT(IS_P2ALIGNED(size, sizeof (uint32_t)));
 
-	fletcher_4_scalar_native(buf, size, zcp);
+#ifndef _KERNEL
+	if (!fletcher_4_initialized)
+		ops = &fletcher_4_scalar_ops;
+	else
+#endif
+		ops = fletcher_4_impl_get();
+
+	if (size >= 1024 && ops->init_incr_native) {
```

Comment:
I'd recommend adding a #define for this 1024 magic value to improve readability.